src/market.py

```diff
@@ -185,6 +190,56 @@ def make_place_order_transaction(self, order_params: NewOrderParams) -> Tuple[Tr
     def find_open_orders_accounts_for_owner(self, owner_address: PublicKey):
         pass
 
+    def cancel_order_by_client_id(self, owner: str) -> str:
+        pass
+
+    def cancel_order(self, owner: Account, order: Order) -> str:
+        transaction = Transaction()
+        transaction.add(self.make_cancel_order_transaction(owner.public_key(), order))
+        return self.send_transaction(transaction, owner)
+
+    def match_orders(self, fee_payer: Account, limit: int) -> str:
+        transaction = Transaction()
+        transaction.add(self.make_match_orders_transaction(limit))
+        return self.send_transaction(transaction, fee_payer)
+
+    def make_cancel_order_transaction(self, owner: PublicKey, order: Order) -> TransactionInstruction:
+        params = CancelOrderParams(
+            market=self.address(),
+            owner=owner,
+            open_orders=order.open_order_address,
+            request_queue=self._decode.request_queue,
+            side=order.side,
+            order_id=order.order_id,
+            open_orders_slot=order.open_order_slot,
+            program_id=self._program_id,
+        )
+        return cancel_order_inst(params)
+
+    def make_match_orders_transaction(self, limit: int) -> TransactionInstruction:
+        params = MatchOrdersParams(
+            market=self.address(),
+            request_queue=PublicKey(self._decode.request_queue),
+            event_queue=PublicKey(self._decode.event_queue),
+            bids=PublicKey(self._decode.bids),
+            asks=PublicKey(self._decode.asks),
+            base_vault=PublicKey(self._decode.base_vault),
+            quote_vault=PublicKey(self._decode.quote_vault),
+            limit=limit,
+            program_id=self._program_id,
+        )
+        return match_order_inst(params)
+
+    def send_transaction(self, transaction: Transaction, *signers: Account) -> str:
+        connection = Client(self._endpoint)
+        res = connection.send_transaction(transaction, *signers, skip_preflight=self._skip_preflight)
+        if self._confirmations > 0:
+            logging.warning("Cannot confirm transaction yet.")
```

Comment:
Can you do something like this for the logger: https://github.com/michaelhly/solana-py/blob/a42801a4dca9c51c31aa4a59a2e733f3b27cd400/solana/rpc/providers/http.py#L22